### PR TITLE
feat(TUP-29701): Upgrade eclipse 418

### PR DESCRIPTION
**Pull request by hzhao-talendbj**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/668/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/668/)

File: core/components-api/src/main/java/org/talend/components/api/component/runtime/JarRuntimeInfo.java [hunk 1]
```diff
@@ -190,6 +189,10 @@ public JarRuntimeInfo(String jarUrlString, String depTxtPath, String runtimeClas
         this(createJarUrl(jarUrlString), depTxtPath, runtimeClassName, SandboxControl.CLASSLOADER_REUSABLE);
     }
 
+    public JarRuntimeInfo(String jarUrlString, String depTxtPath, String runtimeClassName, URLStreamHandler handler) {
```
Comment (wwang-talend) [inline core/components-api/src/main/java/org/talend/components/api/component/runtime/JarRuntimeInfo.java:192]: make it private if only use it in current class

Comment (hzhao-talendbj) [reply core/components-api/src/main/java/org/talend/components/api/component/runtime/JarRuntimeInfo.java]: thanks , I modified it

**Review by wwang-talend — COMMENTED** [review]
Possible to write a junit for it? to show the usage for that.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/670/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/670/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/671/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/671/)

**Review by wwang-talend — APPROVED** [review]

**Review by wwang-talend — CHANGES_REQUESTED** [review]
your added junit failed

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/672/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/672/)

**Review by wwang-talend — APPROVED** [review]

**Pull request merged**